Makefile

```diff
@@ -2,6 +2,7 @@
 sources = pydantic tests docs/build
 isort = isort $(sources)
 black = black -S -l 120 --target-version py38 $(sources)
+mypy = mypy --implicit-optional --no-warn-unused-ignores
```

Comment:
> please update to remove the changes to mypy, otherwise I think I'm happy here but not sure what's related to the original change, and what's related to the mypy chagnes.

the mypy version update is needed to support the recursive signature in [here](https://github.com/pydantic/pydantic/blob/028f2580d4a9cc3179f119fb73ffc32046b69089/pydantic/error_wrappers.py#L47)

As the implicit-optional has been removed as a default setting, removing the `--implicit-optional` parameter creates 37 additional errors I didn't want to suppress via comments
```
PEP 484 prohibits implicit Optional. Accordingly, mypy has changed its default to no_implicit_optional=True
```